### ISSUE TITLE
Add better error messages for otto build

### ISSUE THREAD

**Issue by themccallister** [hashicorp/otto#163]
When trying to test out `otto build` on an existing infrastructure.. I was prompted with the following error message:

```
Error building app: error detecting VCS: no VCS found for path: /Users/jasonmccallister/Desktop/otto-dev
```

It honestly took me a few times to "understand" exactly what the error message was referring too - which was the fact I was not using any version control software.

I do feel like this error message could be written to be a little clearer. Maybe an error message like so:

```
Error building app: error detecting code repository: no VCS found for path: /Users/jasonmccallister/Desktop/otto-dev
```

I realize this is essentially semantics, but its always a win to have useful error messages.

**Comment by Skarlso** [comment]
So, you just exchanged VCS to code repository, which, is the same...?!

**Comment by justincampbell** [comment]
We could explicitly call out Git/SVN/etc here in a more helpful message.

**Comment by themccallister** [comment]
@Skarlso yes its the same, but its generally a bad practice to use acronyms - especially when throwing error messages.

Thanks @justincampbell!

**Comment by mitchellh** [comment]
This is actually a strange error message: what Appfile is causing `otto biuld` to generate this?

**Comment by adambarthelson** [comment]
Not sure if an Appfile is relevant for this case, but here's the one I was using when I ran into this error:

```
application {
  name = "demoProject"
  type = "node"

  dependency { source = "mongodb" }
}

project {
  name = "demoProject"
  infrastructure = "demoProject"
}

infrastructure "demoProject" {
  type = "aws"
  flavor = "simple"
}
```

Initializing Git was all I needed to do to fix it.  Woops!

**Comment by themccallister** [comment]
I was using a PHP application, no Appfile.

It issue was because I did not have a git repo initialized.

**Comment by avinoth** [comment]
I had the same issue with my rails app recently. Not just initializing the repository but only after committing it got resolved.

Error message,

```
Error building app: error getting git commit: exit status 128
stdout: 
stderr: fatal: bad default revision 'HEAD'
```
